### DISCUSSION TITLE
CompletedFileUpload improvements

### DISCUSSION THREAD

**Discussion by Deviad** [micronaut-projects/micronaut-core#5551] [Ideas]
Hello everyone,
currently CompletedFileUpload interface does not offer a delete method. 
For big files or servers that keep running this may represent a problem since the only way to delete these files would be a script that every x hours deletes the files stored in the tmp folder (correct me if I am wrong). 
It would be nice also to have a getPath method in order to be able to work on the same file once it is received rather than creating a copy of it just for the sake of getting the path. For small files this may not be a problem but for files as large as 10 GB this is an issue. 
I know that these things belong to Netty, however maybe there are workarounds for these. 

Thank you. 
Have a nice day,
Davide

**Comment by graemerocher** [comment]
Contributions welcome of course

**Comment by jameskleeh** [comment]
>  currently CompletedFileUpload interface does not offer a delete method.

That is incorrect since Micronaut 2.4 https://github.com/micronaut-projects/micronaut-core/blob/477676f07dbae71ac0178d02a758b5df3d324da8/http/src/main/java/io/micronaut/http/multipart/FileUpload.java#L83

> It would be nice also to have a getPath method in order to be able to work on the same file once it is received 

You are making the assumption the file is stored on the file system. In many cases it is not and thus there is no path

**Reply by jameskleeh** [reply]
If you are receiving big files then I would suggest not using CompletedFileUpload and instead stream the data
